hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -956,7 +959,7 @@ public HdfsFileStatus getFileInfo(String src) throws IOException {
       if (children != null && !children.isEmpty()) {
         Map<String, Long> dates = getMountPointDates(src);
         long date = 0;
-        if (dates != null && dates.containsKey(src)) {
+        if (dates.containsKey(src)) {
```

Comment:
I would leave the null check; it is good practice.

Reply:
Modified it since `getMountPointDates` always return a not null response. I will roll back this modification.